src/Workspaces/CoreTest/EditorConfigStorageLocation/NamingStylePreferenceEditorConfigStorageLocationTests.cs

```diff
@@ -42,38 +33,159 @@ public static void TestEmptyDictionaryDefaultNamingStylePreferencesObjectReturns
         [Fact]
         public static void TestNonEmptyDictionaryReturnsTrue()
         {
-            var initialDictionary = new Dictionary<string, string>()
+            var editorConfigStorageLocation = new NamingStylePreferenceEditorConfigStorageLocation();
+            var newDictionary = new Dictionary<string, string>()
             {
-                ["dotnet_naming_rule.methods_and_properties_must_be_pascal_case.severity"] = "warning",
+                ["dotnet_naming_rule.methods_and_properties_must_be_pascal_case.severity"] = "error",
                 ["dotnet_naming_rule.methods_and_properties_must_be_pascal_case.symbols"] = "method_and_property_symbols",
                 ["dotnet_naming_rule.methods_and_properties_must_be_pascal_case.style"] = "pascal_case_style",
                 ["dotnet_naming_symbols.method_and_property_symbols.applicable_kinds"] = "method,property",
                 ["dotnet_naming_symbols.method_and_property_symbols.applicable_accessibilities"] = "*",
                 ["dotnet_naming_style.pascal_case_style.capitalization"] = "pascal_case"
             };
-            var existingNamingStylePreferences = ParseDictionary(initialDictionary);
 
+            var result = editorConfigStorageLocation.TryGetOption(
+                newDictionary,
+                typeof(NamingStylePreferences),
+                out var combinedNamingStyles);
+
+            Assert.True(result, "Expected non-empty dictionary to return true");
+            var isNamingStylePreferencesObject = combinedNamingStyles is NamingStylePreferences;
```

Comment:
Use `Assert.IsType<NamingStylePreferences>`: not only does it return the casted form, but also will throw a more useful exception that will give the underlying type if it fails (rather than this which lets you fire up a debugger). If you care about derived types, then Assert.IsAssignableFrom.

Reply:
(This also saves you from the code later which is then casting the type too....)

Reply:
It was a copy-paste from the test preceding the new tests. 😄

Reply:
Awesome, thanks. (And that can be deferred if we're taking this through QB and want to do that more urgently of course.)